### PR TITLE
Update to 2018 edition

### PR DESCRIPTION
**Pull request by liuchengxu**
- Update all crates to 2018 edition with some formating. Let me know if you don't want the formatting in this PR.
- Update README example_setup to 2018 edition.

### CODE REVIEW

File: c-marshalling/src/lib.rs [hunk 1]
```diff
@@ -276,8 +272,14 @@ impl<T: IntoRawConversion, E: IntoRawConversion> IntoRawConversion for Result<T,
 
     fn into_raw(self) -> Result<Self::Raw, Error> {
         Ok(match self {
-            Ok(value) => CMutResult { ok: box_into_ptr(value)?, err: ::std::ptr::null_mut() },
-            Err(value) => CMutResult { ok: ::std::ptr::null_mut(), err: box_into_ptr(value)? },
+            Ok(value) => CMutResult {
+                ok: box_into_ptr(value)?,
+                err: ::std::ptr::null_mut(),
```
Comment (gardell) [inline c-marshalling/src/lib.rs]: nit: remove the `::` prefix as well

File: c-marshalling/src/lib.rs [hunk 2]
```diff
@@ -276,8 +272,14 @@ impl<T: IntoRawConversion, E: IntoRawConversion> IntoRawConversion for Result<T,
...
+            },
+            Err(value) => CMutResult {
+                ok: ::std::ptr::null_mut(),
```
Comment (gardell) [inline c-marshalling/src/lib.rs]: same

File: parser/src/lib.rs [hunk 3]
```diff
@@ -87,23 +88,21 @@ pub fn functions(items: &[::syn::Item]) -> Vec<Function> {
...
                 args,
                 ret: match *output {
                     ::syn::ReturnType::Default => quote! { () },
-                    ::syn::ReturnType::Type(_, ref ty) => if let ::syn::Type::Path(ref path) = **ty
-                    {
-                        quote! { #path }
-                    } else {
-                        panic!("Function return type can only be immediate")
-                    },
+                    ::syn::ReturnType::Type(_, ref ty) => {
```
Comment (gardell) [inline parser/src/lib.rs]: same

File: c-marshalling/src/lib.rs (hunk 1 above)
Comment (liuchengxu) [reply c-marshalling/src/lib.rs]: Done, and I should have removed all `::` prefix.

**Comment by scooter-dangle** [comment]
@Marwes, are we okay to merge this since @gardell approved?

**Review by Marwes — APPROVED** [review]

**Pull request merged**